.github/workflows/wheels.yml

```diff
@@ -145,6 +145,15 @@ jobs:
           CIBW_ARCHS: ${{ matrix.buildplat[2] }}
           CIBW_ENVIRONMENT_PASS_LINUX: RUNNER_OS
 
+          # MACOS_DEPLOYMENT_TARGET is set because of
+          # https://github.com/pypa/cibuildwheel/issues/1419. Once that
+          # is closed and meson-python==0.13 is available, then
+          # that environment variable can be removed.
+          CIBW_ENVIRONMENT_MACOS: >
+            MACOSX_DEPLOYMENT_TARGET=10.9
+            MACOS_DEPLOYMENT_TARGET=10.9
+            _PYTHON_HOST_PLATFORM=macosx-10.9-x86_64
```

Comment:
LGTM, I think we may get a point release per https://github.com/mesonbuild/meson-python/pull/309#issuecomment-1430740897, but I agree that making this change is definitely the right solution at the moment and keeps us safe with either variable name.

I checked that the new wheel names for MacOS x86_64 have `10_9` tag--i.e., this log: https://github.com/scipy/scipy/actions/runs/4190628420/jobs/7264230310

We can see that the old wheels used for `1.10.0` wheel build had `10_15` per https://github.com/scipy/scipy/actions/runs/3832354403/jobs/6522562479 (and indeed, per the wheels on PyPI proper at https://pypi.org/project/scipy/1.10.0/#files). NumPy doesn't seem to have that problem for their most recent patch release.

The Windows wheel build failures here are because of https://github.com/scipy/scipy/issues/17934, which is addressed by the backport-labeled PR https://github.com/scipy/scipy/pull/17936, so merging thanks!